tests/e2e/lib/dpa_helpers.go

```diff
@@ -82,23 +85,41 @@ func WithBackupImages(backupImage bool) DpaCROption {
 		return nil
 	}
 }
+func WithBackupLocations(locations []oadpv1alpha1.BackupLocation) DpaCROption {
+	return func(cr *oadpv1alpha1.DataProtectionApplication) error {
+		cr.Spec.BackupLocations = locations
+		return nil
+	}
+}
+
+func WithSnapshotLocations(locations []oadpv1alpha1.SnapshotLocation) DpaCROption {
+	return func(cr *oadpv1alpha1.DataProtectionApplication) error {
+		cr.Spec.SnapshotLocations = locations
+		return nil
+	}
+}
+
+var veleroPrefix = "velero-e2e-" + string(uuid.NewUUID())
+
+func GetVeleroPrefix() string {
+	return veleroPrefix
+}
+
+var dpa *oadpv1alpha1.DataProtectionApplication
```

Comment:
I don't know if we should do it this way. 

Global Singletons are scary, and it is just as scary that you are pass around the reference and I am not seeing any deep copies. 

I wonder if we can re-state the goals of what we want this to do (sounds like some sort of fixtures) and maybe try and come up with a different approach?

Reply:
Would it help if I remove the pointer? I didn't mean to pass reference around. Was just unexporting it. it was `Dpa` before with the same pointer type..
```suggestion
var dpa oadpv1alpha1.DataProtectionApplication
```

Reply:
This var is supposed to be loaded once and for accessing an original copy loaded from JSON per comment in line below it

Reply:
resolved by only allowing access to dpa via getters which returns a copy.

Redesigns will be followed up in another PR.